src/replication.c

```diff
@@ -3135,6 +3374,539 @@ void replicationHandleMasterDisconnection(void) {
+ * │SEND_HANDSHAKE     │        │ │  └───────┬───────────────────────┘                           │
+ * └────────┬──────────┘        │ │          │+OK                                                │
+ *          │+OK                │ │  ┌───────▼───────────────────────┐                           │
+ * ┌────────▼──────────┐        │ │  │ RDB_CH_RECEIVE_REPLCONF_REPLY │                           │
+ * │RECEIVE_AUTH_REPLY │        │ │  └───────┬───────────────────────┘                           │
+ * └────────┬──────────┘        │ │          │+OK                          Main channel state    │
+ *          │+OK                │ │  ┌───────▼───────────────────────┐   ┌──▼────────────────┐   │
+ * ┌────────▼──────────┐        │ │  │ RDB_CH_RECEIVE_FULLRESYNC     │   │REPL_TRANSFER      │ ◄─┼────┐
+ * │RECEIVE_PORT_REPLY │        │ │  └───────┬───────────────────────┘   └──┬────────────────┘   │    │
+ * └────────┬──────────┘        │ │          │+FULLRESYNC ..                │ buffer repl stream │    │
+ *          │+OK                │ │          │                              │ in parallel to RDB │    │
+ * ┌────────▼──────────┐        │ │          │                              │                    │    │
+ * │RECEIVE_IP_REPLY   │        │ │  ┌───────▼───────────────┐              │                    │    │
+ * └────────┬──────────┘        │ │  │REPL_RDB_CH_RDB_LOADING│              │                    │    │
+ *          │+OK                │ │  └───────┬───────────────┘              │                    │    │
+ * ┌────────▼──────────┐        │ │          │Done loading                  │                    │    │
+ * │RECEIVE_CAPA_REPLY │        │ │          │                              │                    │    │
+ * └────────┬──────────┘        │ │          │ Replica loads replication    │                    │    │
+ *          │                   │ │          │ buffer into memory           │                    │    │
+ * ┌────────▼───┐               │ │          └─────────┬────────────────────┘                    │    │
+ * │SEND_PSYNC  │               │ │                    │                                         │    │
+ * └─┬──────────┘               │ │                    │                                         │    │
+ *   │PSYNC (use cached-master) │ │                    │                                         │    │
+ * ┌─▼─────────────────┐        │ │                    │                                         │    │
+ * │RECEIVE_PSYNC_REPLY│        │ └────────────────────┼─────────────────────────────────────────┘    │
+ * └────────┬─┬────────┘        │                      │                                              │
+ * +CONTINUE│ │+RDBCHANNELSYNC <client-id>             │                                              │
+ *   │      │ └─────────────────┘──────────────────────┼──────────────────────────────────────────────┘
```

Comment:
maybe we can put `REPL_TRANSFER ` state here

Reply:
impossible to understand where do you mean :) Below RECEIVE_PSYNC_REPLY?
I've put it to the top right as if like it starts along side with rdb delivery.

Reply:
i saw you changed the diagram, it is what i expect